Source/Engine/PackageBuilder/PackageBuilder.cs

```diff
@@ -146,15 +107,8 @@ private PackageSyntax ParseFile(string filePath)
 
         private LinkedPackageSyntax LinkPackage(PackageSyntax parsedTree, string baseDirectory, string filePath)
         {
-            LinkedPackageSyntax result;
-            var linker = new NormalizingPatternLinker(baseDirectory, requiredPackageLoader: this);
-            fDependencyStack.Push(new FileLinker()
-            {
-                FilePath = filePath,
-                Linker = linker
-            });
-            result = linker.Link(parsedTree);
-            fDependencyStack.Pop();
+            var linker = new NormalizingPatternLinker(ParseFile, fPackageCache);
```

Comment:
Consider passing `fFileContentProvider` to `NormalizingPatternLinker` directly and using
```csharp
var parser = new SyntaxParser();
result = parser.ParsePackageText(text);
```
in two places: `PackageBuilder` and `PatternLinker`.